rbac/management/role/v2_model.py

```diff
@@ -145,6 +145,49 @@ class Meta:
     objects = TypedRoleV2Manager(role_type=_expected_type)
 
 
+class RoleBinding(TenantAwareModel):
+    """A role binding."""
+
+    uuid = models.UUIDField(default=uuid.uuid7, editable=False, unique=True, null=False)
+    role = models.ForeignKey(RoleV2, on_delete=models.CASCADE, related_name="bindings")
+
+    resource_type = models.CharField(max_length=256, null=False)
+    resource_id = models.CharField(max_length=256, null=False)
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=["role", "resource_type", "resource_id"],
+                name="unique role binding per role resource pair",
+            ),
+        ]
+
+
+class RoleBindingPrincipal(models.Model):
+    """The relationship between a RoleBinding and one of its principal subjects."""
+
+    principal = models.ForeignKey(Principal, on_delete=models.CASCADE, related_name="role_binding_entries")
+    binding = models.ForeignKey(RoleBinding, on_delete=models.CASCADE, related_name="principal_entries")
+    source = models.CharField(max_length=128, null=False)
```

Comment:
This is for indicating where the binding is originated from , e.g., which cross account requests. CrossAccountRequest/123 means this principal is bound to the resource because of CAR with id 123. so if there are multiple CARs  123 and 456 and 456 is deleted, we keep the binding.
I wonder if we should make this field a list. So only when source is all gone, then we remove the binding